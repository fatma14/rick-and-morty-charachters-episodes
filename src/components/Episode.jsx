import React, { Component } from "react";

import getEpisodeById from "../service/episodes";
import "./episode.css";

export default class Episode extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      code: "",
      air_date: "",
    };
  }
  componentDidMount() {
    getEpisodeById(this.props.episodeURL).then((response) => {
      this.setState({
        name: response.name,
        air_date: response.air_date,
        code: response.episode,
      });
    });
  }

  render() {
    return (
      <div className="episodeContainer">
        <h3> {this.state.name} </h3>
        <div>
          <b>Episode Code:</b> {this.state.code}{" "}
        </div>
        <div>
          <b>Air Date:</b> {this.state.air_date}{" "}
        </div>
      </div>
    );
  }
}
