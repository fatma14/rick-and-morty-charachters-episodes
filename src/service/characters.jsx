import axios from "axios";

const url = "https://rickandmortyapi.com/api/character/";

export async function getCharacters() {
  const response = await axios.get(url);
  return response.data;
}

export async function getCharacterById(id) {
  const response = await axios.get(url + `${id}`);
  return response.data;
}

export async function getCharactersByName(name) {
  const response = await axios.get(url + `?name=${name}`);
  return response.data;
}

export async function getPage(url) {
  const response = await axios.get(url);
  return response.data;
}
