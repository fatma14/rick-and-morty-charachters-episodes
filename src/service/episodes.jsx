import axios from "axios";

export default async function getEpisodeById(url) {
  const response = axios.get(url);
  return response.data;
}
