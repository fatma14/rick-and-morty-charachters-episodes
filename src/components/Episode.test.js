import React from "react";
import renderer from "react-test-renderer";
import Episode from "./Episode";
import "../service/episodes";

let mockGetEpisodeById;

jest.mock("../service/episodes", () => ({
  __esModule: true,
  default: () => {
    return mockGetEpisodeById;
  },
}));

describe("Episode Unit tests", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot", async () => {
    const episode = {
      name: "test 1",
      air_date: "4th July",
      episodeu: "S01E01",
    };

    mockGetEpisodeById = new Promise((resolve) => {
      resolve(episode);
    });
    let component;
    await renderer.act(async () => {
      component = renderer.create(<Episode />);
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
