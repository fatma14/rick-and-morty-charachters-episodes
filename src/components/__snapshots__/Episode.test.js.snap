// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode Unit tests should match snapshot 1`] = `
<div
  className="episodeContainer"
>
  <h3>
     
    test 1
     
  </h3>
  <div>
    <b>
      Episode Code:
    </b>
     
  </div>
  <div>
    <b>
      Air Date:
    </b>
     
    4th July
  </div>
</div>
`;
